### PR TITLE
layers: Fix vkconfig setting path lookup (Admin)

### LINKED ISSUE #2094: Settings not working (vkconfig admin/IsHighIntegrity)
When running vkconfig and tested application as admin, settings are not picked up by ValidationLayers

Details:

vkconfig tool in SDK v1.2.148 writes path to settings file to HKEY_LOCAL_MACHINE or HKEY_CURRENT_USER windows registry.
Former is used when launched as admin :

```
void Configurator::AddRegistryEntriesForLayers(QString qsJSONFile, QString qsSettingsFile) {
    // Layer override json file
    HKEY key;
    HKEY userKey = (_running_as_administrator) ? HKEY_LOCAL_MACHINE : HKEY_CURRENT_USER;
```

Topic in VulkanTools LunarG/VulkanTools#935
Topic in Loader KhronosGroup/Vulkan-Loader#266

But ValidationLayers always use HKEY_CURRENT_USER:

```
string ConfigFile::FindSettings() {
    struct stat info;

#if defined(WIN32)
    // Look for VkConfig-specific settings location specified in the windows registry
    HKEY hive;
    LSTATUS err = RegOpenKeyEx(HKEY_CURRENT_USER, "Software\\Khronos\\Vulkan\\Settings", 0, KEY_READ, &hive);
    if (err == ERROR_SUCCESS) {
```

ValidationLayers should use the same logic as vkconfig to read settings from registry.

### PR DESCRIPTION
**Pull request by dor666**
Settings lookup for windows registry was hardcoded to HKCU.
Use HKLM when in running as admin.
When running as standard user, additionally check HKCU.

fix #2094

### CODE REVIEW

File: layers/vk_layer_config.cpp [hunk 1]
```diff
@@ -192,37 +192,71 @@ void ConfigFile::SetOption(const string &option, const string &val) {
     value_map_[option] = val;
 }
 
+#if defined(WIN32)
+// Check for admin rights
+static inline bool IsHighIntegrity() {
```
Comment (dor666) [inline layers/vk_layer_config.cpp:197]: Straight copy from loader.

File: layers/vk_layer_config.cpp [hunk 2]
```diff
@@ -192,37 +192,71 @@ void ConfigFile::SetOption(const string &option, const string &val) {
...
-                continue;
-            }
-
-            // Check if this actually points to a file
-            if ((stat(name, &info) != 0) || !(info.st_mode & S_IFREG)) {
-                continue;
+    HKEY key;
+
+    const std::array<HKEY, 2> hives = {DEFAULT_VK_REGISTRY_HIVE, SECONDARY_VK_REGISTRY_HIVE};
+    const size_t hives_to_check_count = IsHighIntegrity() ? 1 : hives.size(); // Admin checks only the default hive
```
Comment (dor666) [inline layers/vk_layer_config.cpp:229]: Wrapped check in loop (1 or 2 iterations).

File: layers/vk_layer_config.h [hunk 3]
```diff
@@ -26,6 +26,13 @@
 #include "vulkan/vk_layer.h"
 #include "vulkan/vulkan.h"
 
+#if defined(WIN32)
```
Comment (dor666) [inline layers/vk_layer_config.h:29]: Copied from loader.

**Comment by mark-lunarg** [comment]
@dor666, running the changes through our internal CI.  This looks great, and while I understand this was copied straight from the Loader, would you mind changing `DWORD `to `uint32_t `please?  It'll help prevent confusion in the future.

**Comment by dor666** [comment]
Thanks,
changed all occurrences DWORD to uint32_t.

**Comment by dor666** [comment]
Unfortunately msvc complains about conversion from &uint32_t to LPDWORD. Should I leave it with those reinterpret casts?

**Comment by mark-lunarg** [comment]
@dor666, I should have known Windows might make this ugly.  Please don't worry about it any more, I'll take the original commit as-is.  I apologize again for the churn!

**Comment by mark-lunarg** [comment]
Passed CI, rebased and pushed the original commit.  Thanks for handling this!

**Comment by dor666** [comment]
Thanks!

**Pull request closed without merging**